File: android/src/main/kotlin/com/cloudwebrtc/webrtc/controller/ChannelNameGenerator.kt
package com.cloudwebrtc.webrtc.controller

/**
 * Generator for the all [io.flutter.plugin.common.MethodChannel] names created by `flutter_webrtc`.
 */
object ChannelNameGenerator {
    private const val PREFIX: String = "FlutterWebRtc"

    /**
     * Generates new channel name for some controller with a provided ID.
     *
     * @param name name of the controller for which we generate new name (e.g. `PeerConnection`).
     * @param id unique identifier of some concrete instance of some entity.
     * @return generated channel name.
     */
    fun name(name: String, id: Long): String {
        return "$PREFIX/$name/$id"
    }
}

File: ios/Classes/controller/ChannelNameGenerator.swift
/// Generator of names for Flutter method/event channels.
class ChannelNameGenerator {
  /// Static prefix for all channels of this plugin
  private static let prefix: String = "FlutterWebRtc"

  /// Last generated ID.
  private static var lastId: Int = 0

  /// Returns new ID for the channel.
  static func nextId() -> Int {
    ChannelNameGenerator.lastId += 1
    return lastId
  }

  /// Returns name for the channel with provided name and ID.
  static func name(name: String, id: Int) -> String {
    "\(ChannelNameGenerator.prefix)/\(name)/\(id)"
  }
}
